### PR TITLE
Sending correct config files per wfms version

### LINKED ISSUE #65: Adjust the Source Code according to Repository Restructure Changes
Adjust the Source Code for the following changes:
- _Data Transformers SUT Configuration_:
  - The file name is always data-transformers.configuration.yml
  - The folder structure, where you can map the `SUT Name` and the `SUT Version` is described in https://github.com/benchflow/sut-plugins/issues/7. This will be placed under `/app/data-transformers/suts/` and `/app/analysers/suts/` respectively. This also impacts https://travis-ci.org/benchflow/data-transformers/builds/143378743 that currently does not build because of the way you reference to the configuration file.
- _Scheduler configuration_:
  - The `data-transformers` and `analysers` configuration file is split in two parts in the respective repos. The name of the file of the data `data-transformers` is `data-transformers.scheduler.configuration.yml`, the one of the `analysers` is `analysers.scheduler.configuration.yml`. They will be placed in the root of the configuration folder.
  - The path of the scripts in the respective configuration file starts from the folder that is in the repo, i.e., it does not include the `/app` part. That is something should be in the scheduler configuration file because it is aware of it's own folders structure.

Refer to commits: [81d58c334651c8431f39ff31d8edf9800819dfa9](https://github.com/benchflow/data-analyses-scheduler/commit/81d58c334651c8431f39ff31d8edf9800819dfa9) and [ff272d5e0705ee4196d22da9d6f01c95f45aea65](https://github.com/benchflow/data-analyses-scheduler/commit/ff272d5e0705ee4196d22da9d6f01c95f45aea65)

Also related to https://github.com/benchflow/data-transformers/issues/62

### PR DESCRIPTION
**Pull request by Cerfoglg**
Fixes #65

TODO:
- [x] Check with #65

<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23issuecomment-235013904%22%2C%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23issuecomment-235027248%22%2C%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72082505%22%2C%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72082715%22%2C%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72082904%22%2C%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72082946%22%2C%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72084273%22%2C%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72086384%22%2C%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72088386%22%2C%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72091585%22%2C%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72091681%22%2C%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72091682%22%2C%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72092663%22%2C%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72095522%22%2C%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72095525%22%2C%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72095531%22%2C%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72104526%22%2C%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72104880%22%2C%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72109681%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23issuecomment-235013904%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40VincenzoFerme%20Also%20using%20app%20path%20in%20configuration%20to%20avoid%20referinc%20/app%20in%20the%20configuration%20where%20it%20doesn%27t%20make%20sense%22%2C%20%22created_at%22%3A%20%222016-07-25T16%3A56%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8819678%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Cerfoglg%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-07-25T17%3A43%3A29Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%203ecea21eb50fe20b535d999e5cb204ce29b67cb4%20src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go%2074%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72082946%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Pass%20it%20using%20the%20configuration%20file%22%2C%20%22created_at%22%3A%20%222016-07-25T15%3A16%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-07-25T15%3A58%3A55Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go%3AL75-94%22%7D%2C%20%22Pull%203ecea21eb50fe20b535d999e5cb204ce29b67cb4%20configuration.yml%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72082505%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%60wfms%60%20is%20a%20SUT%20type%2C%20and%20should%20be%20determined%20using%20the%20test%20configuration.%20The%20%60configurations_path%60%20I%20expect%20should%20be%20%60/app/data-transformers/suts/%60.%20Maybe%20a%20better%20name%20for%20the%20configuration%20variable%20would%20also%20help.%22%2C%20%22created_at%22%3A%20%222016-07-25T15%3A14%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40VincenzoFerme%20And%20where%20do%20we%20say%20that%20we%20are%20dealing%20with%20a%20wfms%3F%20Cause%20right%20now%20the%20only%20thing%20passed%20dynamically%20is%20SUTName%20and%20SUTVersion%2C%20and%20they%20don%27t%20specifiy%20it%22%2C%20%22created_at%22%3A%20%222016-07-25T15%3A22%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8819678%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Cerfoglg%22%7D%7D%2C%20%7B%22body%22%3A%20%22In%20the%20%60benchflow-test.yml%60%20configuration%20file%20you%20also%20have%20the%20%60type%60%20of%20the%20SUT%22%2C%20%22created_at%22%3A%20%222016-07-25T15%3A32%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40VincenzoFerme%20Got%20it%2C%20applied%20changes%22%2C%20%22created_at%22%3A%20%222016-07-25T15%3A41%3A45Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8819678%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Cerfoglg%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-07-25T16%3A20%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20configuration.yml%3AL19-26%22%7D%2C%20%22Pull%203ecea21eb50fe20b535d999e5cb204ce29b67cb4%20src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go%2051%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72082904%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Pass%20it%20using%20the%20configuration%20file%22%2C%20%22created_at%22%3A%20%222016-07-25T15%3A16%3A27Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-07-25T15%3A58%3A55Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go%3AL64-71%22%7D%2C%20%22Pull%205201663b86119117ad6afc4dcd3d6238c9aec505%20src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72091585%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Where%20do%20you%20use%20%60SUTType%60%3F%22%2C%20%22created_at%22%3A%20%222016-07-25T15%3A58%3A25Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40VincenzoFerme%20Sorry%20a%20commit%20didn%27t%20go%20through%2C%20now%20you%20should%20be%20able%20to%20see%20it%22%2C%20%22created_at%22%3A%20%222016-07-25T16%3A04%3A23Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8819678%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Cerfoglg%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-07-25T16%3A20%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go%3AL8-30%22%7D%2C%20%22Pull%20c5af8d516538aabf5027f15028b5b4055cee96ae%20src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go%2028%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72104526%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Don%27t%20you%20need%20%60AppPath%60%20also%20for%20%60dataTransformations.py%60%3F%22%2C%20%22created_at%22%3A%20%222016-07-25T17%3A13%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40VincenzoFerme%20Fixing%22%2C%20%22created_at%22%3A%20%222016-07-25T17%3A16%3A05Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8819678%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Cerfoglg%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-07-25T17%3A43%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go%3AL49-61%22%7D%2C%20%22Pull%203ecea21eb50fe20b535d999e5cb204ce29b67cb4%20src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go%2010%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/benchflow/data-analyses-scheduler/pull/68%23discussion_r72082715%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Add%20comments%20and%20code%20reference%20if%20any%22%2C%20%22created_at%22%3A%20%222016-07-25T15%3A15%3A36Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-07-25T16%3A20%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go%3AL2-34%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-Pull 3ecea21eb50fe20b535d999e5cb204ce29b67cb4 configuration.yml 5'></a> &nbsp;<b><a href='https://github.com/benchflow/data-analyses-scheduler/pull/68#discussion_r72082505'>File: configuration.yml:L19-26</a></b>
- <a href='https://github.com/VincenzoFerme'></a> `wfms` is a SUT type, and should be determined using the test configuration. The `configurations_path` I expect should be `/app/data-transformers/suts/`. Maybe a better name for the configuration variable would also help.
- <a href='https://github.com/Cerfoglg'></a> @VincenzoFerme And where do we say that we are dealing with a wfms? Cause right now the only thing passed dynamically is SUTName and SUTVersion, and they don't specifiy it
- <a href='https://github.com/VincenzoFerme'></a> In the `benchflow-test.yml` configuration file you also have the `type` of the SUT
- <a href='https://github.com/Cerfoglg'></a> @VincenzoFerme Got it, applied changes
- [x] <a href='#crh-comment-Pull 3ecea21eb50fe20b535d999e5cb204ce29b67cb4 src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go 10'></a> &nbsp;<b><a href='https://github.com/benchflow/data-analyses-scheduler/pull/68#discussion_r72082715'>File: src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go:L2-34</a></b>
- <a href='https://github.com/VincenzoFerme'></a> Add comments and code reference if any
- [x] <a href='#crh-comment-Pull 3ecea21eb50fe20b535d999e5cb204ce29b67cb4 src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go 51'></a> &nbsp;<b><a href='https://github.com/benchflow/data-analyses-scheduler/pull/68#discussion_r72082904'>File: src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go:L64-71</a></b>
- <a href='https://github.com/VincenzoFerme'></a> Pass it using the configuration file
- [x] <a href='#crh-comment-Pull 3ecea21eb50fe20b535d999e5cb204ce29b67cb4 src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go 74'></a> &nbsp;<b><a href='https://github.com/benchflow/data-analyses-scheduler/pull/68#discussion_r72082946'>File: src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go:L75-94</a></b>
- <a href='https://github.com/VincenzoFerme'></a> Pass it using the configuration file
- [x] <a href='#crh-comment-Pull 5201663b86119117ad6afc4dcd3d6238c9aec505 src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go 5'></a> &nbsp;<b><a href='https://github.com/benchflow/data-analyses-scheduler/pull/68#discussion_r72091585'>File: src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go:L8-30</a></b>
- <a href='https://github.com/VincenzoFerme'></a> Where do you use `SUTType`?
- <a href='https://github.com/Cerfoglg'></a> @VincenzoFerme Sorry a commit didn't go through, now you should be able to see it
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/benchflow/data-analyses-scheduler/pull/68#issuecomment-235013904'>General Comment</a></b>
- <a href='https://github.com/Cerfoglg'></a> @VincenzoFerme Also using app path in configuration to avoid referinc /app in the configuration where it doesn't make sense
- [x] <a href='#crh-comment-Pull c5af8d516538aabf5027f15028b5b4055cee96ae src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go 28'></a> &nbsp;<b><a href='https://github.com/benchflow/data-analyses-scheduler/pull/68#discussion_r72104526'>File: src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go:L49-61</a></b>
- <a href='https://github.com/VincenzoFerme'></a> Don't you need `AppPath` also for `dataTransformations.py`?
- <a href='https://github.com/Cerfoglg'></a> @VincenzoFerme Fixing

<a href='https://www.codereviewhub.com/benchflow/data-analyses-scheduler/pull/68?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/benchflow/data-analyses-scheduler/pull/68?approve=1'></a>&nbsp;<a href='https://github.com/benchflow/data-analyses-scheduler/pull/68'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by VincenzoFerme** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/benchflow/data-analyses-scheduler/pull/68'>Refresh</a>.<a href='#crh-update-none'></a>

File: configuration.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ alluxio_port: "19998"
 pyspark_cassandra_version: "0.3.5"
 analysers_path: "/app/analysers"
 transformers_path: "/app/data-transformers"
-configurations_path: "/app/configuration"
+configurations_path: "/app/data-transformers/suts/wfms"
```
Comment (VincenzoFerme) [inline configuration.yml]: `wfms` is a SUT type, and should be determined using the test configuration. The `configurations_path` I expect should be `/app/data-transformers/suts/`. Maybe a better name for the configuration variable would also help.

File: src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go [hunk 2]
```diff
@@ -2,9 +2,32 @@ package scripts
 
 import (
 	"encoding/json"
+	"io/ioutil"
+	"strings"
 	. "cloud/benchflow/data-analyses-scheduler/vars"
 )
 
+// Function to resolve the correct configuration file path based on the SUT Name and Version
+func getConfigFilePath(SUTVersion string, SUTName string, fileName string) string {
```
Comment (VincenzoFerme) [inline src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go]: Add comments and code reference if any

File: src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go [hunk 3]
```diff
@@ -38,8 +64,7 @@ func ConstructTransformerSubmitArguments(s TransformerScript, msg KafkaMessage,
 	transformerArguments.Minio_host = MinioHost
 	transformerArguments.Minio_port = MinioPort
 	transformerArguments.Minio_secret_key = MinioSecretKey
-	transformerArguments.SUT_Name = SUTName
-	transformerArguments.SUT_Version = SUTVersion
+	transformerArguments.Config_file = "data-transformers.configuration.yml"
```
Comment (VincenzoFerme) [inline src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go]: Pass it using the configuration file

File: src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go [hunk 4]
```diff
@@ -50,16 +75,19 @@ func ConstructTransformerSubmitArguments(s TransformerScript, msg KafkaMessage,
...
 	args = append(args, "--py-files", AnalysersPath+"/commons/commons.py,"+SparkHome+"/pyspark-cassandra-assembly-"+PysparkCassandraVersion+".jar")
 	args = append(args, script)
 	analyserArguments := AnalyserArguments{}
 	analyserArguments.Cassandra_keyspace = CassandraKeyspace
 	analyserArguments.Container_ID = containerID
 	analyserArguments.Experiment_ID = experimentID
 	analyserArguments.Host_ID = hostID
-	analyserArguments.SUT_Name = SUTName
-	analyserArguments.SUT_Version = SUTVersion
+	analyserArguments.Config_file = "analysers.configuration.yml"
```
Comment (VincenzoFerme) [inline src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go]: Pass it using the configuration file

File: configuration.yml (hunk 1 above)
Comment (Cerfoglg) [inline configuration.yml]: @VincenzoFerme And where do we say that we are dealing with a wfms? Cause right now the only thing passed dynamically is SUTName and SUTVersion, and they don't specifiy it

Comment (VincenzoFerme) [inline configuration.yml]: In the `benchflow-test.yml` configuration file you also have the `type` of the SUT

Comment (Cerfoglg) [inline configuration.yml]: @VincenzoFerme Got it, applied changes

File: src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go [hunk 5]
```diff
@@ -8,20 +8,22 @@ import (
 )
 
 // Function to resolve the correct configuration file path based on the SUT Name and Version
-func getConfigFilePath(SUTVersion string, SUTName string, fileName string) string {
+func getConfigFilePath(SUTVersion string, SUTName string, SUTType string, fileName string) string {
```
Comment (VincenzoFerme) [inline src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go:11]: Where do you use `SUTType`?

File: src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go (hunk 3 above)
Comment (VincenzoFerme) [inline src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go]: :+1:<a href='#crh-update-none'></a>

File: src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go (hunk 4 above)
Comment (VincenzoFerme) [inline src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go]: :+1:<a href='#crh-update-complete'></a>

File: src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go (hunk 5 above)
Comment (Cerfoglg) [inline src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go:11]: @VincenzoFerme Sorry a commit didn't go through, now you should be able to see it

File: configuration.yml (hunk 1 above)
Comment (VincenzoFerme) [inline configuration.yml]: :+1:<a href='#crh-update-none'></a>

File: src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go [hunk 6]
```diff
@@ -2,9 +2,34 @@ package scripts
 
 import (
 	"encoding/json"
+	"io/ioutil"
+	"strings"
```
Comment (VincenzoFerme) [inline src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go:6]: :+1:<a href='#crh-update-none'></a>

File: src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go (hunk 2 above)
Comment (VincenzoFerme) [inline src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go]: :+1:<a href='#crh-update-complete'></a>

**Comment by Cerfoglg** [comment]
@VincenzoFerme Also using app path in configuration to avoid referinc /app in the configuration where it doesn't make sense

File: src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go [hunk 7]
```diff
@@ -49,12 +49,12 @@ func constructSparkArguments() []string {
 func ConstructTransformerSubmitArguments(s TransformerScript, msg KafkaMessage, containerID string, hostID string, SUTName string, SUTVersion string, SUTType string) []string {
 	var args []string
 	args = constructSparkArguments()
-	args = append(args, "--py-files", TransformersPath+"/commons/commons.py"+","+TransformersPath+"/transformations/dataTransformations.py"+","+SparkHome+"/pyspark-cassandra-assembly-"+PysparkCassandraVersion+".jar")
+	args = append(args, "--py-files", AppPath+"/"+TransformersPath+"/commons/commons.py"+","+TransformersPath+"/transformations/dataTransformations.py"+","+SparkHome+"/pyspark-cassandra-assembly-"+PysparkCassandraVersion+".jar")
```
Comment (VincenzoFerme) [inline src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go]: Don't you need `AppPath` also for `dataTransformations.py`?

Comment (Cerfoglg) [inline src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go]: @VincenzoFerme Fixing

**Comment by VincenzoFerme** [comment]
:+1:<a href='#crh-update-none'></a>

File: src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go [hunk 8]
```diff
@@ -2,9 +2,34 @@ package scripts
...
+		}
+    	vRange := strings.Split(dirName, "-")
+    	if len(vRange) == 2 {
+    		vNumsRangeLow := strings.Split(vRange[0], ".")
+    		vNumsRangeHigh := strings.Split(vRange[1], ".")
+    		if versionNums[0] >= vNumsRangeLow[0] && versionNums[0] <= vNumsRangeHigh[0] && versionNums[1] >= vNumsRangeLow[1] && versionNums[1] <= vNumsRangeHigh[1] && versionNums[2] >= vNumsRangeLow[2] && versionNums[2] <= vNumsRangeHigh[2] {
+    			return AppPath+"/"+TransformersConfigurationsPath+"/"+SUTType+"/"+SUTName+"/"+dirName+"/"+fileName
+    		}
+		}
+    }
```
Comment (VincenzoFerme) [inline src/cloud/benchflow/data-analyses-scheduler/scripts/sparkArgsConstructors.go:29]: :+1:<a href='#crh-update-complete'></a>

**Pull request merged**